[25기 전태양] [Mission 4]로그인,메인 레이아웃 필수구현 및 로그인 유효성 검사 완료

xodid157 opened the pull request:
- 로그인 페이지 레이아웃 구현 완료
-i d,pw 입력 시 로그인 버튼 활성화 기능 구현 완료
- 로그인 버튼 활성화 구현 완료
- 메인 페이지 레이아웃 구현 완료
- 로그인 유효성 검사 구현 완료

## 수정 사항 간략한 한줄 요약
(여기에 수정 사항에 대한 간략한 한줄 요약/제목 작성해 주세요.)
로그인,메인 레이아웃 필수구현 및 로그인 유효성 검사 완료
		 
## 수정 사항들 자세한 내용
(여기에 수정 사항에 대한 자세한 내용을 작성해 주세요.)
- id '@' 포함, pw 5 글자 이상 입력 시 버튼 활성화
- 로그인 화면 enter 누르면 버튼 클릭 구현 완료
- 로그인 화면에 유효하지 않은 값이면  '아이디 또는 비밀번호가 잘못 입력 되어있습니다. ' 출력 구현
- 메인화면 피드 2개 넣고 2개다 댓글 기능 구현 완료


## 기타 질문 및 특이 사항
(궁금한 사항들, 하면서 느낀 점들 공유해주실 것이 있으면 자유롭게 작성해 주세요.)
css에서 반복되는 애들을 common에 넣지 못한거 가고 클래스 명이 부자연 스러운거 같습니다.

## 체크 리스트 (아래 사항들이 전부 체크되어야만 merge가 됩니다!)
- [x] 필요한 test들을 완료하였고 기능이 제대로 실행되는지 확인 하였습니다.
- [x] Wecode의 코드 스타일 가이드에 맞추어 코드를 작성 하였습니다.
- [x] 제가 의도한 파일들과 수정 사항들만 커밋이 된 것을 확인 하였습니다.
- [x] 본 수정 사항들을 팀원들과 사전에 상의하였고 팀원들 모두 해당 PR에 대하여 알고 있습니다.

dipsiiiiiiiiii commented on students/11th/yoonmincho/js/login.js:
```diff
@@ -0,0 +1,25 @@
+const pwd = document.getElementsByClassName('input22')[0]
+const button = document.getElementsByClassName('loginbutton')[0]
+
+id.addEventListener('keyup',function(){
+  if( id.value && pwd.value ) { 
+    button.style.backgroundColor = '#0095f6';
+    console.log('hello');
+  }
+  else {
+    button.style.backgroundColor = 'rgba(0,149,246,.3)';
+  }
+})
+
+pwd.addEventListener('keyup',function(){
+  if( id.value && pwd.value ) { 
+    button.style.backgroundColor = '#0095f6';
+    console.log('hi');
+  }
+  else {
+    button.style.backgroundColor = 'rgba(0,149,246,.3)';
+  }
+})                        
```
줄일 수 있는 방법은 
1. 이벤트 함수를 따로 빼서 정의해서 사용하면 됩니다.

이렇게 함수를 따로 뺀 후에 
`
function handleBtnColor(){
if( id.value && pwd.value ) { 
    button.style.backgroundColor = '#0095f6';
    console.log('hi');
  }
  else {
    button.style.backgroundColor = 'rgba(0,149,246,.3)';
  }
}
`
이렇게 줄일 수 있습니다.
`
id.addEventListener('keyup',handleBtnColor);
pwd.addEventListener('keyup',handleBtnColor);
`

dipsiiiiiiiiii replied:
여기서 추가로 줄일 수 있는 방법은 forEach를 사용해서
`
[id, pwd].forEach(el => {
  el.addEventListener('keyup', handleBtnColor)
}
`
이렇게 할 수는 있는데 여기서는 2줄 밖에 안되서 별로 차이가 없어 보이는데 여러 인풋이었으면 훨씬 효과적입니다! 이렇게 줄이는 방법은 꼭 아셨으면 좋겠네요!

dipsiiiiiiiiii commented on students/11th/yoonmincho/login.html:
```diff
@@ -0,0 +1,40 @@
+<!DOCTYPE html>
+<html>
+
```
태그 사이의 불필요한 빈 칸은 모두 지워주세요!

yoonminsohappy replied:
수정했습니다

dipsiiiiiiiiii commented on students/11th/yoonmincho/login.html:
```diff
@@ -0,0 +1,40 @@
+ <div class="container">
+    <div class="loginbox">
+
+     <img class="logo" src="img/logo_text.png">
```
img 태그는 alt 속성을 적어서 어떤 이미지인지 설명해줄 필요가 있습니다. 모든 img 태그에 alt 꼭 넣어주세요. 그리고 src 속성보다 앞에 오게 적어주세요.

yoonminsohappy replied:
수정했습니다

dipsiiiiiiiiii commented on students/11th/yoonmincho/login.html:
```diff
@@ -0,0 +1,40 @@
+
+     <img class="logo" src="img/logo_text.png">
+
+     <div class="input1">
```
클래스 이름은 말씀드렸다시피 클래스 이름만 봐도 어떤 레이아웃인지, 어떤 역할을 하는지 알 수 있게 적어주세요!

yoonminsohappy replied:
수정했습니다

dipsiiiiiiiiii commented on students/11th/yoonmincho/login.html:
```diff
@@ -0,0 +1,40 @@
+  <meta
+    name="viewport                                                                                                                 "
+    content="width=device-width">
+  <title>repl.it</title>
```
타이틀도 한 번 바꿔보세요! 탭 이름이 바뀝니다.

yoonminsohappy replied:
수정했습니다

dipsiiiiiiiiii commented on students/11th/yoonmincho/main.html:
```diff
@@ -0,0 +1,135 @@
+  <body>
+
+    <nav>
+       <div class="navleftcontainer">
```
클래스명도 camelCase!

yoonminsohappy replied:
로그인페이지는 수정했습니다

yoonminsohappy replied:
메인페이지도 수정했습니다

dipsiiiiiiiiii commented on students/11th/yoonmincho/style/main.css:
```diff
@@ -0,0 +1,261 @@
+    height: 2000px;
+    display: flex;
+    flex-direction: row;
+    margin: 30px auto;
+    padding-top: 30px;
```
padding을 주기 보다는 margin의 30px을 60px로 바꾸는게 더 좋을 것 같습니다!

yoonminsohappy replied:
수정했습니다

dipsiiiiiiiiii commented on students/11th/yoonmincho/style/main.css:
```diff
@@ -0,0 +1,261 @@
+    width: 285px;
+    height: 600px;
+    position: fixed;
+    margin-left: 650px;
```
feed 오른쪽으로 보내기 위해서 margin-left를 주셨는데 이렇게 하기 보다는
두 요소를 가로로 나란히 정렬하는 거라서 display: flex로 주시는 게 더 빠르고 편합니다! flex에 익숙해지셔야 합니다!

dipsiiiiiiiiii commented on students/11th/yoonmincho/style/main.css:
```diff
@@ -0,0 +1,261 @@
+.dots {
+    width: 24px;
+    height: 24px;
+    margin-left: 450px;   
```
여기서도 margin-left로 오른쪽에 이동시키지 말고 flex를 활용해보세요

dipsiiiiiiiiii commented on students/11th/yoonmincho/main.html:
```diff
@@ -0,0 +1,135 @@
+
+      <div class="recommendBox">
+
+        <div class="eachRecommendList">
```
직접 적으신 클래스명처럼 list니까 li태그를 사용해주세요. ul로 감싸고!

xodid157 committed:
-로그인 레이아웃 완료
-메인화면 레이아웃 완료
-로그인 버튼 활성화 구현 완료

xodid157 committed:
- 로그인 화면 레이아웃 구현완료
- 로그인 화면 아이디 비밀번호 하나이상씩 들어올때 버튼 활성화 완료
- 메인화면 레이아웃 구현 완료
- 메인화면 댓글 기능 구현 완료

xodid157 committed:
add: 로그인,메인 레이아웃및 필수 기술구현 완료
- 로그인 화면 레이아웃 완료
- 로그인 화면 id,ps 입력 및 유효성 검사 enter 치면 버튼활성화 구현완료
-메인 화면 레이아웃 구현완료
-댓글 추가 기능 구현 완료

xodid157 committed:
메인 댓글인풋창 변경

xodid157 committed:
css조금 수정
